pkg/operator/ceph/object/objectstore.go

```diff
@@ -1000,3 +999,38 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// check whether zone endpoint list need to be updated or not
+func UpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) bool {
```

Comment:
This function is unnecessarily complicated, and it is hard to read (and therefore maintain). There is a high degree of what is called "cyclomatic complexity" in this function. Please break nested `for` loops.

Having several variables to indicate true/false values for returning is unnecessary. We should be able to return true/false immediately on recognizing a particular condition and return the opposite (false/true) at the end when we are sure it's not the "condition."